editor/src/components/canvas/canvas-strategies/strategies/grid-rearrange-move-strategy.ts

```diff
@@ -127,21 +128,128 @@ export const gridRearrangeMoveStrategy: CanvasStrategyFactory = (
         ),
       ]
 
+      const { commands, patch } =
+        strategyToApply.type === 'GRID_REARRANGE'
+          ? getCommandsAndPatchForGridRearrange(
+              canvasState,
+              interactionSession.interactionData,
+              customState,
+              selectedElement,
+            )
+          : getCommandsAndPatchForReparent(
```

Comment:
shouldn't at this point this strategy become disabled, and the reparent metastrategy take over?

Reply:
as I noted in the description, there are a bunch of problems that currently prevent it:
1. grid interactions have their own drag logic
2. absolute reparent needs absolute props, which when lacking would not kick in the strategy altogether

The current code just triggers the strategy logic. There's obviously some duplication but it felt like a good compromise.

With that said, if I'm missing something I'm more than open to other approaches

Reply:
oh right, that sounds like the good tradeoff then